doc/source/f2py/signature-file.rst

```diff
@@ -234,172 +241,188 @@ Other statements:
 
         entry <entry name> [([<arguments>])]
 
-      F2PY generates wrappers to all entry names using the signature
+      F2PY generates wrappers for all entry names using the signature
       of the routine block.
 
-      Tip: ``entry`` statement can be used to describe the signature
-      of an arbitrary routine allowing F2PY to generate a number of
-      wrappers from only one routine block signature. There are few
-      restrictions while doing this: ``fortranname`` cannot be used,
-      ``callstatement`` and ``callprotoargument`` can be used only if
-      they are valid for all entry routines, etc.
+      .. note::
+
+        Tip: The ``entry`` statement can be used to describe the signature
+        of an arbitrary routine allowing F2PY to generate a number of
+        wrappers from only one routine block signature. There are few
```

Comment:
The meaning or "routine" is unclear here. Maybe "subroutine" or some such?

Reply:
The `entry` statement is a bit of an odd one. AFAIK it isn't used often; and it works on the level of a `subprogram`. Basically an `entry` statement associates an additional procedure and it is typically used in external procedures... 

They're detailed in `15.6.2.6` of the [J3/18-007r1 F2018 interpretation document](https://gitlab.com/lfortran/lfortran/uploads/973db2bfcc64cd2a397f823c32718b39/18-007r1.pdf) which also has a nice visual:

![image](https://user-images.githubusercontent.com/4336207/134053214-ee110bf1-f073-4774-a32b-4b801da16f95.png)


They're essentially superseded by internal functions from F90 onwards. (internal subroutines and functions cannot contain `entry` statements).

However, I see the issue with the readability of this. Essentially it was a way to define additional `functions` and `subroutines` from the subprogram scope before internal functions and subroutines were provided.

I think I could change this to arbitrary subroutine or function since that would be the main `f2py` use case.

Reply:
OK, let's leave that for another PR.